x-pack/plugins/security_solution/public/timelines/store/middlewares/timeline_changed.test.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createMockStore } from '../../../common/mock';
+import { selectTimelineById } from '../selectors';
+import { TimelineId } from '../../../../common/types/timeline';
+
+import { setChanged, updateKqlMode, showTimeline } from '../actions';
+
+jest.mock('../actions', () => {
+  const actual = jest.requireActual('../actions');
+  return {
+    ...actual,
+    setChanged: jest.fn().mockImplementation((...args) => actual.setChanged(...args)),
+  };
+});
+
+const setChangedMock = setChanged as unknown as jest.Mock;
+
+describe('Timeline changed middleware', () => {
+  let store = createMockStore();
+
+  beforeEach(() => {
+    store = createMockStore();
+    setChangedMock.mockClear();
+  });
+
+  it('should mark a timeline as changed for some actions', () => {
```

Comment:
This may be overkill, but do we want to test all the actions that could cause it to be changed? I just see a scenario where something is accidentally removed and we don't end up catching it. Maybe you export `timelineChangedTypes` and test those values match some inline snapshot or something?

It would've been nice if we had mocked actions.ts so that you could just call those and verify change is only triggered for the ones we expect..





```

Reply:
It will be hard to test all available actions in Kibana if they mark a timeline as changed 😄 

I went with this approach that tests the `known` actions: https://github.com/elastic/kibana/pull/178009/commits/22a04b29be7882ee09c82a243690bd71edef5f5f

Reply:
💪🏾 nice work, thanks!